integration/plugin/mountpoint/mountpoint_unix_test.go

```diff
@@ -0,0 +1,888 @@
+	}
+	ctrl[1].attachRes = ctrl[0].attachRes
+
+	id, err := request.RunV(client, []string{"/:/host"}, "busybox", []string{"top"})
+	require.Nil(c, err)
+
+	// restart without attached plugins
+	d.Restart(c, "--live-restore")
+
+	err = request.Stop(client, id)
+	require.Nil(c, err)
+
+	// uninitialized plugins are initialized before detach
+	require.Equal(c, []string{"0:properties", "1:properties", "0:attach", "1:attach", "1:properties", "1:detach", "0:properties", "0:detach"}, events)
+
+	events = []string{}
+	id, err = request.RunV(client, []string{"/:/host"}, "busybox", []string{"top"})
+	require.Nil(c, err)
+	defer request.Stop(client, id)
+
+	// no new plugin events have occurred without explicit plugin loading
+	require.Equal(c, []string{}, events)
+}
+
+func TestMountPointPluginContainerLabel(c *testing.T) {
+	defer setupTest(c)()
+	d.StartWithBusybox(c,
+		fmt.Sprintf("--mount-point-plugin=%s5", testMountPointPlugin))
+	client, err := d.NewClient()
+	require.Nil(c, err)
+
+	ctrl[5].attachRes = mountpoint.AttachResponse{
+		Success: false,
+		Err:     "only containers with 'allow' label are allowed to mount",
+	}
+
+	_, err = request.RunV(client, []string{"/:/host"}, "busybox", []string{"true"})
+	require.NotNil(c, err)
+	require.True(c, strings.HasSuffix(err.Error(), fmt.Sprintf("Error response from daemon: middleware plugin:%s5 failed with error: %s: only containers with 'allow' label are allowed to mount", testMountPointPlugin, mountpoint.MountPointAPIAttach)), err.Error())
+
+	_, err = request.RunVLabel(client, []string{"/:/host"}, map[string]string{"allow": ""}, "busybox", []string{"true"})
+	require.Nil(c, err)
+}
+
+func TestMountPointPluginEmergencyDetach(c *testing.T) {
+	defer setupTest(c)()
+	d.StartWithBusybox(c,
+		fmt.Sprintf("--mount-point-plugin=%s0", testMountPointPlugin))
+	client, err := d.NewClient()
+	require.Nil(c, err)
+
+	pwd, err := os.Getwd()
+	require.Nil(c, err)
+	targetFile := filepath.Join(pwd, "target")
+
+	ctrl[0].attachRes = mountpoint.AttachResponse{
+		Success: true,
+		Attachments: []mountpoint.Attachment{
+			{
+				Attach: true,
+				EmergencyDetach: []mountpoint.EmergencyDetachAction{
+					{Remove: targetFile},
+					{Warning: "ran emergency detach"},
+				},
+			},
+		},
+	}
+
+	id, err := request.RunV(client, []string{pwd + ":/host"}, "busybox",
+		[]string{"sh", "-c", "touch /host/target && top"})
+	require.Nil(c, err)
+
+	file, err := os.Open(targetFile)
+	require.Nil(c, err)
+	file.Close()
+
+	err = server[0].Config.Close()
+	require.Nil(c, err)
+	defer setupPlugin(0)
+
+	err = request.Stop(client, id)
+	require.Nil(c, err)
+
+	_, err = os.Open(targetFile)
+	require.True(c, os.IsNotExist(err))
+}
+
+func memoryLimitSupport(t *testing.T) {
+	if !sysInfo.MemoryLimit {
+		t.Skip("no memory limit support")
+	}
+}
+
+func swapMemorySupport(t *testing.T) {
+	if !sysInfo.SwapLimit {
+		t.Skip("no swap limit support")
+	}
+}
+
+func daemonIsLinux(t *testing.T) {
+	if testEnv.DaemonInfo.OSType != "linux" {
```

Comment:
Instead of hiding the checks in these helpers I think it's more obvious to use `gotestyourself/skip`:

(in the test case):
```go
skip.IfCondition(t, testEnv.DaemonInfo.OSType != "linux")
```

same for the two above.